plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/io/sxfm/SXFMFormat.java

```diff
@@ -457,14 +457,17 @@ private void buildFeatureTree(BufferedReader reader) throws UnsupportedModelExce
 					feat.getFeature().getStructure().setParent(lastFeat.getFeature().getStructure());
 					lastFeat.getFeature().getStructure().addChild(feat.getFeature().getStructure());
 					lastFeat = feat;
-					if (lineText.contains("[1,1]")) {
-						lastFeat.getFeature().getStructure().setAlternative();
-					} else if (lineText.contains("[1,*]")) {
-						lastFeat.getFeature().getStructure().setOr();
-					} else if ((lineText.contains("[")) && (lineText.contains("]"))) {
-						final int index = lineText.indexOf('[');
-						final int start = Character.getNumericValue(lineText.charAt(index + 1));
-						final int end = Character.getNumericValue(lineText.charAt(index + 3));
+					if ((lineText.contains("[")) && (lineText.contains("]"))) {
+						final String substring = lineText.substring(lineText.indexOf('[') + 1, lineText.indexOf(']'));
+						final String[] numbers = substring.split(",");
+						final int start = Integer.parseInt(numbers[0]);
+						int end = 0;
+						if (numbers[1].equals("*")) {
```

Comment:
Are we sure that `numbers` always has two fields?

Reply:
We only accept a group definition of the following form: `:g [A,B]`. `A` can only be a number while `B` can be a number or the `*`. Other cases are not accepted when parsing. So yes more or less than two number results in an exception that is marked for the user.

Reply:
See other reply